test/e2e_node/user_namespaces_test.go

```diff
@@ -0,0 +1,97 @@
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2enode
+
+import (
+	"context"
+
+	"github.com/onsi/ginkgo/v2"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/uuid"
+	"k8s.io/kubernetes/pkg/features"
+	kubeletconfig "k8s.io/kubernetes/pkg/kubelet/apis/config"
+	"k8s.io/kubernetes/test/e2e/framework"
+	e2eoutput "k8s.io/kubernetes/test/e2e/framework/pod/output"
+	admissionapi "k8s.io/pod-security-admission/api"
+	"k8s.io/utils/ptr"
+)
+
+var _ = SIGDescribe("User Namespaces [Serial] [Feature:UserNamespacesSupport] [Feature:UserNamespacesPodSecurityStandards] [LinuxOnly]", func() {
+	f := framework.NewDefaultFramework("user-namespaces-test")
+	f.NamespacePodSecurityEnforceLevel = admissionapi.LevelRestricted
+
+	ginkgo.Context("with UserNamespacesSupport and UserNamespacesPodSecurityStandards enabled", func() {
+		tempSetCurrentKubeletConfig(f, func(ctx context.Context, cfg *kubeletconfig.KubeletConfiguration) {
+			defer withFeatureGate(features.UserNamespacesSupport, true)()
+			defer withFeatureGate(features.UserNamespacesPodSecurityStandards, true)()
+			cfg.FeatureGates[string(features.UserNamespacesSupport)] = true
+			cfg.FeatureGates[string(features.UserNamespacesPodSecurityStandards)] = true
+		})
+
+		newPod := func() *v1.Pod {
+			name := "pod-user-namespaces-" + string(uuid.NewUUID())
+
+			return &v1.Pod{
+				ObjectMeta: metav1.ObjectMeta{Name: name},
+				Spec: v1.PodSpec{
+					RestartPolicy:   v1.RestartPolicyNever,
+					HostUsers:       ptr.To(false),
+					SecurityContext: &v1.PodSecurityContext{},
+					Containers: []v1.Container{
+						{
+							Name:    name,
+							Image:   busyboxImage,
+							Command: []string{"echo", "test"},
+							SecurityContext: &v1.SecurityContext{
+								AllowPrivilegeEscalation: ptr.To(false),
+								Capabilities:             &v1.Capabilities{Drop: []v1.Capability{"ALL"}},
+								SeccompProfile:           &v1.SeccompProfile{Type: v1.SeccompProfileTypeRuntimeDefault},
+							},
+						},
+					},
+				},
+			}
+		}
+
+		ginkgo.It("should allow pod with RunAsUser", func(ctx context.Context) {
+			pod := newPod()
+
+			user := ptr.To(int64(1000))
+			pod.Spec.SecurityContext.RunAsUser = user
+			pod.Spec.Containers[0].SecurityContext.RunAsUser = user
+
+			e2eoutput.TestContainerOutput(ctx, f, "RunAsUser", pod, 0, []string{"test"})
+		})
+
+		ginkgo.It("should allow pod with RunAsNonRoot", func(ctx context.Context) {
```

Comment:
These two tests seem redundant, both are testing it can run as UID 1000. The second one is setting the runAsNonRoot to true, but the base pod we are using already has that...

Am I missing something?

Reply:
Moved it into a single test.